### PR TITLE
feat(libs): create EligibilityManager library

### PR DESCRIPTION
**Pull request by biancadanforth**
Because:

* Plan eligibility depends on a number of factors, including what plans the user is already subscribed to (in Stripe, Google IAP or Apple IAP) and available upgrade paths for those plans per product-specific config (in Stripe metadata currently but soon to live in Contentful).
* We want to break out eligibility checks and types into a separate library from the auth server's `CapabilityService.getPlanEligibility`, since it's not really capability-oriented.

This commit:

* Generates a new nx library in lib/payments/eligibility

Closes #FXA-7582

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: libs/payments/eligibility/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (biancadanforth) [inline libs/payments/eligibility/.eslintrc.json:1]: Lisa's [PR](https://github.com/mozilla/fxa/pull/15791) for the `CapabilityManager` library has an `eslintrc.json` file in her `capability` directory as does my [shared error library PR](https://github.com/mozilla/fxa/pull/15627).

My diff from running the [`nx g` command in the Ecosystem docs](https://mozilla.github.io/ecosystem-platform/reference/npm-scripts-and-nx#creating-a-library-in-nx) did not, and I manually added it.

@xlisachan or @julianpoy : Is there something missing from this command, or has something changed to the library generation config in the meantime?

File: libs/payments/eligibility/project.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      "executor": "@nx/linter:eslint",
+      "outputs": ["{options.outputFile}"],
+      "options": {
+        "lintFilePatterns": [
+          "libs/payments/eligibility/**/*.ts",
+          "libs/payments/eligibility/package.json"
+        ]
+      }
+    },
+    "test-unit": {
```
Comment (biancadanforth) [inline libs/payments/eligibility/project.json:29]: When I ran the test command per the generated README, `nx test-unit payments-eligibility`, I got a code coverage warning from the generated `jest.config.ts` file[1] -- Is this expected? Are we doing something with code coverage thus far?

[1]:
```
Validation Warning:

  Unknown option "coverageDirectory" with value "../../../coverage/libs/payments/eligibility" was found.
  This is probably a typing mistake. Fixing it will remove this message.

  Configuration Documentation:
  https://jestjs.io/docs/configuration
```

**Review by biancadanforth — COMMENTED** [review]
Added a couple of comments in-line for the review.

File: libs/payments/eligibility/.eslintrc.json (hunk 1 above)
Comment (biancadanforth) [reply libs/payments/eligibility/.eslintrc.json:1]: Ah nevermind; I see it was generated. I must have accidentally deleted it afterward. Thanks Lisa for the sanity check!

File: libs/payments/eligibility/project.json (hunk 2 above)
Comment (pdehaan) [reply libs/payments/eligibility/project.json:29]: /cc @Trinaa @ashrivastava-qa re: exciting code coverage convo

Comment (julianpoy) [reply libs/payments/eligibility/project.json:29]: This is expected at the moment. The error can safely be ignored.

File: libs/payments/eligibility/.eslintrc.json (hunk 1 above)
Comment (biancadanforth) [reply libs/payments/eligibility/.eslintrc.json:1]: Okay actually a related question here: while eslintrc.json was added, it was added originally with a `@nx/dependency-checks` rule below resulting in a Lint job error also below:

`eslintrc.json` addition I since removed after talking with @xlisachan who had previously chatted with Ben  -- @bbangert : What's this rule about, why is it getting automatically added, why don't we want it to be added, and can we adjust the library gen config to avoid adding it if we don't need it?:
```
{
"files": ["*.json"],
  "parser": "jsonc-eslint-parser",
  "rules": {
    "@nx/dependency-checks": "error"
  }
}
```

Linting error with above eslint rule applied:
```
bdanforth ~/Projects/fxa (FXA-7582) $ npx nx run payments-eligibility:lint

> nx run payments-eligibility:lint


Linting "payments-eligibility"...
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.1.0

YOUR TYPESCRIPT VERSION: 5.2.2

Please only submit bug reports when using the officially supported version.

=============

/Users/bdanforth/Projects/fxa/libs/payments/eligibility/package.json
  1:1  error  Definition for rule '@nx/dependency-checks' was not found  @nx/dependency-checks

✖ 1 problem (1 error, 0 warnings)

Lint errors found in the listed files.


 ———————————————————————————————————————————————————————————————————————————

 >  NX   Ran target lint for project payments-eligibility (1s)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
 
   View structured, searchable error logs at https://nx.app/runs/brWydsctWx
```

File: libs/payments/eligibility/project.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      },
+      "configurations": {
+        "ci": {
+          "ci": true,
+          "codeCoverage": true
+        }
+      }
+    }
+  },
+  "tags": []
```
Comment (sardesam) [inline libs/payments/eligibility/project.json:44]: Similar to `CapabilityManager`, do we have to update this line to:

`"tags": ["scope:shared:lib:payments"]`

File: libs/payments/eligibility/eslintrc.json
Comment (sardesam) [inline libs/payments/eligibility/eslintrc.json:1]: Is this file necessary?

File: libs/payments/eligibility/tsconfig.lib.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "module": "commonjs",
```
Comment (sardesam) [inline libs/payments/eligibility/tsconfig.lib.json:4]: Is this line necessary as it is already included in `./tsconfig.json`?

**Review by sardesam — CHANGES_REQUESTED** [review]
Excellent work! I have a few questions inline that I want to resolve before approving, and noticed you are still working through a couple things as seen from the other comments.

Let me know if you have any questions and/or want to pair. Would be happy to jump on a call! 🙂

File: libs/payments/eligibility/eslintrc.json
Comment (biancadanforth) [reply libs/payments/eligibility/eslintrc.json:1]: It's part of what's generated by `nx` via `nx g`. It's been part of all the previous nx library PRs to date. Currently it doesn't have any overrides, but it does specify a base eslint config, so I imagine yes, we do want it? Since we want at least base linting rules applied to these library files.

Comment (sardesam) [reply libs/payments/eligibility/eslintrc.json:1]: There's a `.eslintrc.json` file but this is the only `eslintrc.json` file I can see in the codebase.

File: libs/payments/eligibility/project.json (hunk 3 above)
Comment (biancadanforth) [reply libs/payments/eligibility/project.json:44]: Hmm I can; this isn't mentioned in the docs for generating nx libraries... What do we plan to use these tags for? Referencing: https://nx.dev/reference/project-configuration#tags . I can add it for sure since it seems to be something we are adding after the library is generated.

File: libs/payments/eligibility/.eslintrc.json (hunk 1 above)
Comment (biancadanforth) [reply libs/payments/eligibility/.eslintrc.json:1]: FWIW answering this question shouldn't block the PR IMO as Lisa previously discussed this with Ben for her PR (that was already merged) and it's a matter of understanding why. I plan to ask Ben when he returns tomorrow (and read a bit more about it beforehand).

File: libs/payments/eligibility/eslintrc.json
Comment (biancadanforth) [reply libs/payments/eligibility/eslintrc.json:1]: Sorry! Yes I removed the `eslintrc.json`. I manually added that in error.

**Review by sardesam — APPROVED** [review]
r+ !
